pyclang/idf_extension.py

```diff
@@ -1,9 +1,28 @@
 import os.path
+import sys
 
 from pyclang import Runner
 
 
 def action_extensions(base_actions, project_path):
+    def get_toolchain(args):
+        cache_path = os.path.join(args.build_dir, 'CMakeCache.txt')
+        if not os.path.exists(cache_path):
+            return None
+        with open(cache_path, encoding='utf-8') as f:
+            for line in f:
+                if line.startswith('IDF_TOOLCHAIN:STRING='):
+                    return line.split('=', maxsplit=1)[1].strip()
+        return None
+
+    def check_clang_toolchain(subcommand_name, args):
+        toolchain = get_toolchain(args)
+        if not toolchain or toolchain == 'clang':
+            return
+        print((f'WARNING: clang actions "{subcommand_name}" is intended for use '
```

Comment:
Should this be `clang action XXX is ...` ? (not `actions`)

Reply:
Yeah, thanks for catching this. I will fix it.